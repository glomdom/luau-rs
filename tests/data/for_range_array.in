fn main() {
    let arr = [1, 2, 3];

    for x in arr {}
}