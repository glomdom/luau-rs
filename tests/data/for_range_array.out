Block(
    Block {
        statements: [
            Function(
                Function {
                    name: "main",
                    params: [],
                    ret_type: None,
                    body: Block(
                        Block {
                            statements: [
                                Let(
                                    Let {
                                        name: "arr",
                                        expr: Array(
                                            Array {
                                                elements: [
                                                    Value(
                                                        Value {
                                                            value: "1",
                                                        },
                                                    ),
                                                    Value(
                                                        Value {
                                                            value: "2",
                                                        },
                                                    ),
                                                    Value(
                                                        Value {
                                                            value: "3",
                                                        },
                                                    ),
                                                ],
                                            },
                                        ),
                                    },
                                ),
                                ForIn(
                                    ForIn {
                                        vars: [
                                            "x",
                                        ],
                                        iter: Value(
                                            Value {
                                                value: "arr",
                                            },
                                        ),
                                        body: Block(
                                            Block {
                                                statements: [],
                                            },
                                        ),
                                    },
                                ),
                            ],
                        },
                    ),
                },
            ),
        ],
    },
)